### ISSUE TITLE
My experimental results are inconsistent with those presented in the paper.

### ISSUE THREAD

**Issue by will-bug** [HaojunYu1998/UltraDet#12]
My experimental results are inconsistent with those presented in the paper.
The results I have reproduced are as follows:
{"Recall": {"R@16": 0.9563803847195882}, "Precision": {"P@R0.7": 0.9492963790893555, "P@R0.8": 0.9196462035179138, "P@R0.9": 0.8418611705303192}, "Average Precision": {"mAP": 0.5496285213430383, "AP50": 0.9148865938186646, "AP75": 0.49401530623435974}, "FP stat": {"FP@R0.7": 3.273577552611068, "FP@R0.8": 7.014809041309431, "FP@R0.9": 21.979734996102884}}
The command used for execution was: --num-gpus 1 --machine-rank 6 --eval-only --resume --config-file configs/Base-UltraDet_R34_C4.yaml DATASETS.SPLIT test DATASETS.NUM_FOLDS 1 MODEL.WEIGHTS pretrained_models/ultradet.pth

During execution, an error is triggered at the location below. Consequently, I have commented out the statement, as there are instances where fanno["ann"] is an empty list ([]).
assert fanno["img"]["video_id"] == fanno["ann"][0]["video_id"]

**Comment by HaojunYu1998** [comment]
Thanks for your issue. We will try to reproduce your results and bugs using your hyper-parameters. And we will let you know if there are any updates.

**Comment by xjtulyc** [comment]
Thank you very much for trying to reproduce our work and for providing such detailed information about the issues you encountered!

Regarding the two problems you've raised:

**1. About the assert error and empty annotation list:**

The situation where `fanno["ann"]` is an empty list is actually expected, as it typically occurs when certain frames don't have any lesion annotations. Commenting out this assert statement is indeed a reasonable solution. We will fix this issue in the code by adding appropriate conditional checks to handle such cases.

**2. About the inconsistent experimental results:**

Your results show:
- AP50: 0.9149 (vs. 0.916 in paper)
- P@R0.9: 0.8419 (vs. 0.908 in paper)
- FP@R0.9: 21.98 (vs. 5.7 in paper)

There is indeed a notable discrepancy. Possible reasons include:

a) **Dataset version issue**: Please confirm that you are using our provided high-quality annotation files (test.json and trainval.json), and that the video classifications (benign/malignant) in rawframes have been updated according to the latest corrections from the CVA-Net authors. Differences in data organization structure can significantly affect the results.

b) **Evaluation settings**: Please verify:
   - Whether you're using the correct test set split
   - Whether NUM_FOLDS is set correctly
   - Whether image preprocessing matches our settings

c) **Environment differences**: Although the impact is usually minor, different PyTorch/CUDA versions may lead to slight numerical variations.

**Suggested troubleshooting steps:**
1. Please double-check that your dataset organization structure exactly matches what's described in the README
2. Verify the number of annotations and videos in test.json
3. Ensure all pretrained weight files are loaded correctly

If the problem persists, could you provide some statistics about your dataset (such as the number of videos in the test set, total number of frames, etc.)? This would help us better locate the issue.

Thank you again for your feedback - it's very helpful for improving our code quality!

Best regards
